transport/http/outbound.go

```diff
@@ -126,49 +126,14 @@ func (o *Outbound) Call(ctx context.Context, treq *transport.Request) (*transpor
 	deadline, _ := ctx.Deadline()
 	ttl := deadline.Sub(start)
 
-	p, err := o.getPeerForRequest(ctx, treq)
-	if err != nil {
-		return nil, err
-	}
-	endRequest := p.StartRequest()
-	defer endRequest()
-
-	req, err := o.createRequest(p, treq)
-	if err != nil {
-		return nil, err
+	response, span, err := o.call(ctx, treq, start, ttl)
+	if span != nil {
+		defer span.Finish()
 	}
-
-	req.Header = applicationHeaders.ToHTTPHeaders(treq.Headers, nil)
-	ctx, req, span := o.withOpentracingSpan(ctx, req, treq, start)
-	defer span.Finish()
-	req = o.withCoreHeaders(req, treq, ttl)
-
-	client, err := o.getHTTPClient(p)
 	if err != nil {
 		return nil, err
 	}
 
-	response, err := client.Do(req.WithContext(ctx))
-
-	if err != nil {
-		// Workaround borrowed from ctxhttp until
-		// https://github.com/golang/go/issues/17711 is resolved.
-		select {
-		case <-ctx.Done():
-			err = ctx.Err()
-		default:
-		}
-
-		span.SetTag("error", true)
-		span.LogEvent(err.Error())
-		if err == context.DeadlineExceeded {
-			end := time.Now()
-			return nil, errors.ClientTimeoutError(treq.Service, treq.Procedure, end.Sub(start))
-		}
-
-		return nil, err
-	}
-
 	span.SetTag("http.status_code", response.StatusCode)
```

Comment:
IMO we should add the status code tot he span for all HTTP requests. I would
move this `SetTag` to `call`. This will also allow you to `defer span.Finish`
in `call`.